cookies/views/resource.py

```diff
@@ -673,6 +741,51 @@ def resource_content(request, resource_id):
     return HttpResponse('Nope')    # TODO: say something more informative!
 
 
+@auth.authorization_required(ResourceAuthorization.VIEW, _get_resource_by_id)
+def resource_text_content(request, resource_id):
+    """
+    Serve up the raw text content associated with a :class:`.Resource`\.
+
+    This is not the most efficient way to serve files, but we need some kind of
+    security layer here for non-public content.
+    """
+
+    parent_resource = _get_resource_by_id(request, resource_id)
+    content_relations = parent_resource.content.filter(is_deleted=False, content_type='text/plain')
+    if not content_relations:
+        return HttpResponse('No text content found for resource \'%s\'.' % parent_resource.name)
+
+    content_relation = content_relations[0]
```

Comment:
if there can be more than one then this should probably also be part of the request (defaulting to the first).